staging/vhost-device-console/src/vhu_console.rs

```diff
@@ -0,0 +1,1441 @@
+                            }
+                        }
+                    }
+                }
+            }
+            Ok(())
+        })
+    }
+
+    /// Start console thread.
+    pub(crate) fn start_console_thread(
+        vhu_console: &Arc<RwLock<VhostUserConsoleBackend>>,
+    ) -> JoinHandle<Result<()>> {
+        let vhu_console = Arc::clone(vhu_console);
+        trace!("Enter text and press Enter: ");
+
+        let exit_eventfd = vhu_console.read().unwrap().exit_event.as_raw_fd();
+        // Spawn a new thread to handle input.
+        spawn(move || {
+            let term = Term::stdout();
+            let mut fdset = FdSet::new();
+            fdset.insert(term.as_raw_fd());
+            fdset.insert(exit_eventfd);
+            let max_fd = fdset.highest().expect("Failed to read fdset!") + 1;
+
+            loop {
+                let ready = vhu_console.read().unwrap().ready_to_write;
+                let exit = vhu_console.read().unwrap().controller.read().unwrap().exit;
+
+                if exit {
+                    dbg!("Exit!");
+                    break;
+                } else if ready {
+                    let mut fdset_clone = fdset;
+                    enable_raw_mode().expect("Raw mode error");
+
+                    match select(Some(max_fd), Some(&mut fdset_clone), None, None, None) {
+                        Ok(_num_fds) => {
+                            disable_raw_mode().expect("Raw mode error");
+                            let exit = vhu_console.read().unwrap().controller.read().unwrap().exit;
+                            if (fdset_clone.contains(exit_eventfd)) && exit {
+                                dbg!("Exit!");
+                                break;
+                            }
+
+                            if fdset_clone.contains(term.as_raw_fd()) {
+                                let character =
+                                    match term.read_key().expect("Error reading character!") {
+                                        Key::Char(character) => Some(character),
+                                        Key::Enter => Some('\n'),
+                                        Key::Tab => Some('\t'),
+                                        Key::Backspace => Some('\u{8}'),
+                                        _ => None,
+                                    };
+
+                                if character.is_some() {
+                                    // Pass the data to vhu_console and trigger an EventFd
+                                    let input_buffer =
+                                        character.expect("Unexpected character!").to_string();
+                                    let _ =
+                                        vhu_console.write().unwrap().rx_data_fifo.add(input_buffer);
+                                    vhu_console.write().unwrap().rx_event.write(1).unwrap();
+                                }
```

Comment:
```suggestion
                                if let Some(character) = match term.read_key().unwrap() {
                                    Key::Char(character) => Some(character),
                                    Key::Enter => Some('\n'),
                                    Key::Tab => Some('\t'),
                                    Key::Backspace => Some('\u{8}'),
                                    _ => None,
                                } {
                                    // Pass the data to vhu_console and trigger an EventFd
                                    let input_buffer = character.to_string();
                                    vhu_console
                                        .write()
                                        .unwrap()
                                        .rx_data_fifo
                                        .add(input_buffer)
                                        .unwrap();
                                    vhu_console.write().unwrap().rx_event.write(1).unwrap();
                                }
```
I'd prefer this

Reply:
@dorindabassey, thanks for the review comments. I think all of them have been addressed. Let me know if you have any additional comments to discuss